Source/Fuse.Nodes/PreviewState.uno

```diff
@@ -0,0 +1,107 @@
+using Uno.Collections;
+
+namespace Fuse
+{
+	/** 
+		State management in Preview for reloading/reifying. This has not use other than when running in preview.
+	*/
+	class PreviewState
+	{
+		public PreviewStateData Save()
+		{
+			var psd = new PreviewStateData();
+			for (int i=0; i < _savers.Count; ++i)
+				_savers[i].Save( psd );
+			return psd;
+		}
+		
+		public void SetState(PreviewStateData data)
+		{
+			if (data != null && data.Consumed)
+			{
+				Fuse.Diagnostics.InternalError( "Attempting to restore an already consumed state", this );
+				_current = null;
+				return;
+			}
+			
+			_current = data;
+		}
+		
+		PreviewStateData _current;
+		public PreviewStateData Current
+		{
+			get { return _current; }
+		}
+		
+		List<IPreviewStateSaver> _savers = new List<IPreviewStateSaver>();
+		
+		public void AddSaver( IPreviewStateSaver saver )
+		{
+			_savers.Add( saver );
+		}
+		
+		public void RemoveSaver( IPreviewStateSaver saver )
+		{
+			_savers.Remove( saver );
+		}
+		
+		/** Finds the appropriate PreviewState for a given node. May return null */
+		static public PreviewState Find( Node n )
+		{
+			while (n != null)
+			{
+				var rv = n as RootViewport; //the only thing that provides a PreviewState
+				if (rv != null)
+					return rv.PreviewState;
+				n = n.Parent;
+			}
+			return null;
+		}
+	}
+	
+	class PreviewStateData
+	{
+		/** Has this state data already been consumed/restored? */
+		public bool Consumed;
```

Comment:
Although there is a check for this above, I can't see it being set anywhere

Reply:
Yeah, this seems to be unused.

Reply:
Yeah, this seems to be unused. Please remove :)

Reply:
It's not meant to be unused. I forget to set it. It's a safety check.

Reply:
@mortoray: Yeah, looks good.